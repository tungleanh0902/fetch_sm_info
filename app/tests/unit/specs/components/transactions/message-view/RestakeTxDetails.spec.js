import { shallowMount } from "@vue/test-utils"
import RestakeTxDetails from "src/components/transactions/message-view/RestakeTxDetails"

describe(`RestakeTxDetails`, () => {
  let wrapper

  const tx = {
    type: "RestakeTx",
    hash: "3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    height: 308453,
    timestamp: "2020-01-10T09:02:54Z",
    memo: "",
    success: true,
    fees: [
      {
        denom: "ATOM",
        amount: "0.00045",
      },
    ],
    details: {
      from: ["cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"],
      to: ["cosmosvaloper18ymm350peujvq2xy9ymyqj4v34ekvnk3tsekuz"],
      amount: {
        denom: "ATOM",
        amount: "10",
      },
    },
  }

  it(`renders a restake transaction message`, () => {
    wrapper = shallowMount(RestakeTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
