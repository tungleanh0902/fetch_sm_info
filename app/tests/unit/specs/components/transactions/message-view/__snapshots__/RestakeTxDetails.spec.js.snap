// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestakeTxDetails renders a restake transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Restaked"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Restaked
    </h3>
     
    <span>
      From 
    </span>
     
    <router-link-stub
      to="/staking/validators/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      <!---->
      
      cosmosvaloper…9yf2 
    </router-link-stub>
    <i
      class="material-icons notranslate arrow"
    >
      arrow_right_alt
    </i>
     
    <router-link-stub
      to="[object Object]"
    >
      <!---->
      
      cosmosvaloper…ekuz
    
    </router-link-stub>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10 
      ATOM
    
    </p>
  </div>
</div>
`;
